internal/controller/oauth/types.go

```diff
@@ -0,0 +1,32 @@
+package oauth
+
+import (
+	"context"
+	"time"
+)
+
+// FlowType represents different OAuth/OIDC flow types
+type FlowType string
+
+const (
+	FlowClientCredentials            FlowType = "client_credentials"
+	FlowClientCredentialsWithIDToken FlowType = "client_credentials_with_id_token"
+)
+
+// TokenResponse represents the common token response structure
+type TokenResponse struct {
+	AccessToken  string
+	TokenType    string
+	ExpiresAt    time.Time
+	Scope        string
+	IDToken      string // Optional OIDC field
+	RefreshToken string // Optional refresh token
+	Raw          map[string]interface{}
+}
+
+// Provider defines the interface for OAuth token providers
+type Provider interface {
```

Comment:
@missBerg I think the Provider interface is confusing, from the code we are not actually implementing different oauth2/oidc providers like GitHub, google etc if that was what this is meant. It is implementing the OIDC protocol itself.